core/controllers/suggestion_test.py

```diff
@@ -768,6 +773,101 @@ def test_translation_suggestion_creation_with_new_images(self):
+
+        suggestion = suggestion_services.query_suggestions(
+            [('author_id', self.translator_id), ('target_id', self.EXP_ID)])[0]
+        self.assertEqual(
+            suggestion.change.translation_html, '<p>Updated In Hindi</p>')
+        self.logout()
+
+    def test_update_suggestion_already_accepted_suggestion_exception(self):
+        self.login(self.ADMIN_EMAIL)
+        change_dict = {
+            'cmd': 'add_translation',
+            'content_id': 'content',
+            'language_code': 'hi',
+            'content_html': '<p>old content html</p>',
+            'state_name': 'State 1',
+            'translation_html': '<p>Translation for content.</p>'
+        }
+
+        suggestion = suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            feconf.ENTITY_TYPE_EXPLORATION,
+            'exp1', 1, self.translator_id, change_dict, 'description')
+        suggestion_services.accept_suggestion(
+            suggestion.suggestion_id, self.reviewer_id, 'Accepted', 'Done'
+        )
+
+        csrf_token = self.get_new_csrf_token()
+        response = self.put_json('%s/%s' % (
+            feconf.UPDATE_SUGGESTION_URL_PREFIX,
+            suggestion.suggestion_id), {
+                'change': {
+                    'cmd': 'add_translation',
+                    'content_id': 'content',
+                    'language_code': 'hi',
+                    'content_html': '<p>old content html</p>',
+                    'state_name': 'State 1',
+                    'translation_html': '<p>Translation for content.</p>'
+                }
+            }, csrf_token=csrf_token, expected_status_int=400)
+        self.assertEqual(
+            response['error'],
+            'The suggestion with id %s has been accepted or rejected' % (
+                suggestion.suggestion_id))
+        self.logout()
+
+    def test_update_suggestion__change_parameter_missing_exception(self):
+        self.login(self.ADMIN_EMAIL)
+        change_dict = {
+            'cmd': 'add_translation',
+            'content_id': 'content',
+            'language_code': 'hi',
+            'content_html': '<p>old content html</p>',
+            'state_name': 'State 1',
+            'translation_html': '<p>Translation for content.</p>'
+        }
+
+        suggestion = suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            feconf.ENTITY_TYPE_EXPLORATION,
+            'exp1', 1, self.translator_id, change_dict, 'description')
+
+        csrf_token = self.get_new_csrf_token()
+        response = self.put_json(
+            '%s/%s' % (
+                feconf.UPDATE_SUGGESTION_URL_PREFIX,
+                suggestion.suggestion_id), {},
+            csrf_token=csrf_token,
+            expected_status_int=400)
+        self.assertEqual(
+            response['error'],
+            'The parameter \'change\' is missing.')
+        self.logout()
```

Comment:
I added State.validate() in UpdateQuestionSuggestionHandler to validate the question_state_data object.

Reply:
Sure, but where's the backend test?

Reply:
I added a test for an invalid question_state_data with an empty object. Does that make sense?

Reply:
Oh, I see. I was thinking that some sub-part of it would be invalid, but I think what you have is fine too. Can you rename the variable to invalid_question_state_data to make it a bit clearer?

Reply:
Done